contracts/verifiers/UniversalVerifier.sol

```diff
@@ -74,6 +83,21 @@ contract UniversalVerifier is
         );
     }
 
+    function updateZKPRequest(
```

Comment:
Some unit tests should be there for the upgrade and event check

Reply:
Nice catch! Added unit tests.